streams/src/test/java/org/apache/kafka/streams/processor/internals/RecordCollectorTest.java

```diff
@@ -101,7 +103,7 @@
 
     private final MockProducer<byte[], byte[]> mockProducer = new MockProducer<>(
         cluster, true, new DefaultPartitioner(), byteArraySerializer, byteArraySerializer);
-    private final StreamsProducer streamsProducer = new StreamsProducer(mockProducer, false, logContext);
+    private final StreamsProducer streamsProducer = new StreamsProducer(mockProducer, AT_LEAST_ONCE, logContext);
```

Comment:
This test is about `RecrodCollector` and most test use the "at-least-once" `StreamsProducer`. In fact, there are only three special test for eos-alpha and those would be the same for eos-beta. Not sure if we would gain anything?

Reply:
For the test completeness, we could optionally duplicate the cases for EOS-alpha and beta just to make sure the passing in processing-mode is taking effect.

Reply:
I was thinking about this, but it would be redundant tests, because the `RecordCollectorImpl` only calls `StreamsProducer.eosEnabled()` that return `true` for eos-alpha and eos-beta and thus the `RecordCollector` cannot distinguish both cases anyway. (And we have unit test in `StreamsProducerTest` to make sure that `eosEnabled()` works correctly.)